index.js

```diff
@@ -195,24 +195,29 @@ function PgSimplifyInflectorPlugin(
           ? proc.tags.fieldName + ListSuffix(proc)
           : this.camelCase(pseudoColumnName + listSuffix(proc));
       },
-      singleRelationByKeys(detailedKeys, table, _foreignTable, constraint) {
+      singleRelationByKeys(detailedKeys, table, foreignTable, constraint) {
         if (constraint.tags.fieldName) {
           return constraint.tags.fieldName;
         }
         const baseName = this.getBaseNameFromKeys(detailedKeys);
         if (baseName) {
           return this.camelCase(baseName);
         }
-        if (this.baseNameMatches(baseName, table.name)) {
+        if (this._isPrimaryKey(detailedKeys, foreignTable)) {
           return this.camelCase(`${this._singularizedTableName(table)}`);
         }
         return oldInflection.singleRelationByKeys(
           detailedKeys,
           table,
-          _foreignTable,
+          foreignTable,
           constraint
         );
       },
+      _isPrimaryKey(detailedKeys, table) {
+        return detailedKeys.every(
+          (key, i) => key === table.primaryKeyConstraint.keyAttributes[i]
+        );
+      },
```

Comment:
This is not an inflector; please extract it out of the inflectors object to the top scope as a utility function.

Reply:
Fixed.